warp10/src/main/java/io/warp10/script/functions/SNAPSHOT.java

```diff
@@ -230,10 +234,14 @@ public static void addElement(SNAPSHOT snapshot, StringBuilder sb, Object o, boo
         sb.append(" ");
       } else if (o instanceof String) {
         sb.append("'");
-        try {
-          sb.append(WarpURLEncoder.encode(o.toString(), "UTF-8"));
-        } catch (UnsupportedEncodingException uee) {
-          throw new WarpScriptException(uee);
+        if (readable) {
+          sb.append(o);
```

Comment:
This does not work, if the String has special characters they will be displayed as is and the resulting WarpScript will be invalid.

You could emit the string with ' and anything below 32 URL encoded, but not the raw string.